EpiAware/src/EpiObsModels/LatentDelay.jl

```diff
@@ -0,0 +1,60 @@
+@doc raw"
+The `LatentDelay` struct represents an observation model that introduces a latent delay in the observations. It is a subtype of `AbstractObservationModel`.
+
+## Fields
+- `model::M`: The underlying observation model.
+- `pmf::T`: The probability mass function (PMF) representing the delay distribution.
+
+## Constructors
+- `LatentDelay(model::M, distribution::C; D = 15, Δd = 1.0) where {M <: AbstractObservationModel, C <: ContinuousDistribution}`: Constructs a `LatentDelay` object with the given underlying observation model and continuous distribution. The `D` parameter specifies the number of discrete delay intervals, and the `Δd` parameter specifies the width of each delay interval.
+
+- `LatentDelay(model::M, pmf::T) where {M <: AbstractObservationModel, T <: AbstractVector{<:Real}}`: Constructs a `LatentDelay` object with the given underlying observation model and delay PMF.
+
+## Examples
+```julia
+using Distributions, Turing, EpiAware
+obs = LatentDelay(NegativeBinomialError(), truncated(Normal(5.0, 2.0), 0.0, Inf))
+obs_model = generate_observations(obs, missing, fill(10, 10))
+rand(obs_model)
+```
+"
+struct LatentDelay{M <: AbstractObservationModel, T <: AbstractVector{<:Real}} <:
+       AbstractObservationModel
+    model::M
+    pmf::T
+
+    function LatentDelay(model::M, distribution::C; D = 15,
+            Δd = 1.0) where {M <: AbstractObservationModel, C <: ContinuousDistribution}
+        pmf = create_discrete_pmf(distribution; Δd = Δd, D = D)
+        return LatentDelay(model, pmf)
+    end
+
+    function LatentDelay(model::M,
+            pmf::T) where {M <: AbstractObservationModel, T <: AbstractVector{<:Real}}
+        @assert all(pmf .>= 0) "Delay interval must be non-negative"
+        @assert isapprox(sum(pmf), 1) "Delay interval must sum to 1"
+        new{typeof(model), typeof(pmf)}(model, pmf)
+    end
+end
+
+@doc raw"
+Generates observations based on the `LatentDelay` observation model.
+
+## Arguments
+- `obs_model::LatentDelay`: The `LatentDelay` observation model.
+- `y_t`: The current observations.
+- `I_t`: The current infection indicator.
+
+## Returns
+- `y_t`: The updated observations.
+- `obs_aux`: Additional observation-related variables.
+
+"
+@model function EpiAwareBase.generate_observations(obs_model::LatentDelay, y_t, Y_t)
+    kernel = generate_observation_kernel(obs_model.pmf, length(Y_t))
```

Comment:
We've discussed this a bit f2f, as per the top:

> It also enforces our design goal of not having structs know about the specific index of the problem and so moves the creation of the sparse delay kernel into the generate_obserations call.

I think its reasonable to have a pattern where `obs_model` is expected to know how to construct the kernel if its dispatching on a method that defines a behaviour where it doesn't change with different parameters.

e.g. 

```julia
generate_observation_kernel(obs_model::StaticDelay) #<-- just gets a pre-calculated kernel
```

This is probably a new issue?

Reply:
Yes good idea I think this might be a new issue related to #161 (i.e resolve that and then this is a specific instance?)

Reply:
Are you happy to make one as you might be best placed to flesh out above?

Reply:
Once we have an issue for this and this convo is resolved this PR will merge if we are happy?